Content.Client/UserInterface/Systems/ProgressColorSystem.cs

```diff
@@ -0,0 +1,83 @@
+using Content.Shared.CCVar;
+using Robust.Shared.Configuration;
+
+namespace Content.Client.UserInterface.Systems;
+
+/// <summary>
+/// This system handles getting an interpolated color based on the value of a cvar.
+/// </summary>
+public sealed class ProgressColorSystem : EntitySystem
+{
+    [Dependency] private readonly IConfigurationManager _configuration = default!;
+
+    private bool _colorBlindFriendly;
+
+    private static readonly Color[] Plasma =
+    {
+        new(240, 249, 33),
+        new(248, 149, 64),
+        new(204, 71, 120),
+        new(126, 3, 168),
+        new(13, 8, 135)
+    };
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        Subs.CVar(_configuration, CCVars.AccessibilityColorblindFriendly, OnColorBlindFriendlyChanged, true);
+    }
+
+    private void OnColorBlindFriendlyChanged(bool value, in CVarChangeInfo info)
+    {
+        _colorBlindFriendly = value;
+    }
+
+    public Color GetProgressColor(float progress)
+    {
+        if (!_colorBlindFriendly)
+        {
+            if (progress >= 1.0f)
+            {
+                return new Color(0f, 1f, 0f);
+            }
+
+            // lerp
+            var hue = 5f / 18f * progress;
+            return Color.FromHsv((hue, 1f, 0.75f, 1f));
+        }
+
+        return InterpolateColorGaussian(Plasma, progress);
+    }
+
+    /// <summary>
+    /// Interpolates between multiple colors based on a gaussian distribution.
+    /// Taken from https://stackoverflow.com/a/26103117
+    /// </summary>
+    public static Color InterpolateColorGaussian(Color[] colors, double x)
```

Comment:
Also since these math ops are very expensive, I just realized you can significantly optimize this function by removing the first loop to calculate `total` and just calculate the result with total after the loop.

Reply:
afaik you need the complete total calculated for the second loop or else the percentage calculations on lines 76-78 will be skewed towards the early entries in the list. For example, in the first iteration, `total` will equal `percent` and the blending will be off.

Reply:
Do the division by total *after* the loop. It's constant inside the loop, so what I'm suggesting is taking:

```
A / total + B / total + C / total
```

and turning it into
```
(A + B + C) / total
```

which is identical